File: sol/solution/0001-0100/0006. Zigzag Conversion/Solution.dart
class Solution {
  String convert(String s, int numRows) {
    if (numRows == 1 || s.length <= numRows) return s;
    List<StringBuffer> rows = List.generate(numRows, (_) => StringBuffer());
    int curRow = 0;
    bool goingDown = false;
    for (var c in s.split('')) {
      rows[curRow].write(c);
      if (curRow == 0 || curRow == numRows - 1) goingDown = !goingDown;
      curRow += goingDown ? 1 : -1;
    }
    return rows.map((sb) => sb.toString()).join('');
  }
}

File: sol/solution/0001-0100/0006. Zigzag Conversion/Solution.swift
class Solution {
    func convert(_ s: String, _ numRows: Int) -> String {
        if numRows == 1 || s.count <= numRows { return s }
        var rows = Array(repeating: "", count: min(numRows, s.count))
        var curRow = 0
        var goingDown = false
        for c in s {
            rows[curRow] += String(c)
            if curRow == 0 || curRow == numRows - 1 { goingDown.toggle() }
            curRow += goingDown ? 1 : -1
        }
        return rows.joined()
    }
}